airflow/metrics/otel_logger.py

```diff
@@ -154,9 +201,33 @@ def gauge(
         delta: bool = False,
         *,
         tags: Attributes = None,
+        back_compat_name: str = "",
     ) -> None:
-        warnings.warn("OpenTelemetry Gauges are not yet implemented.")
-        return None
+        """
+        Record a new value for a Gauge.
+
+        :param stat: The name of the stat to update.
+        :param value: The new value of stat, either a float or an int.
+        :param rate: value between 0 and 1 that represents the sample rate at
+            which the metric is going to be emitted.
+        :param delta: If true, the provided value will be added to the previous value.
+            If False the new value will override the previous.
+        :param tags: Tags to append to the stat.
+        :param back_compat_name:  If an alternative name is provided, the
+            stat will be emitted using both names if possible.
+        """
+        if rate < 0:
+            raise ValueError("rate must be a positive value.")
+        if rate < 1 and random.random() > rate:
+            return
```

Comment:
Probably worth factoring this out into a helper instead of reimplementing it twice (soon to be three with timers)